vllm/utils.py

```diff
@@ -1088,3 +1114,69 @@ async def _run_task_with_lock(task: Callable, lock: asyncio.Lock, *args,
     """Utility function to run async task in a lock"""
     async with lock:
         return await task(*args, **kwargs)
+
+
+def _create_dummy_modules():
+    import importlib
+    import types
+
+    habana_frameworks = types.ModuleType('habana_frameworks')
+    spec = importlib.util.spec_from_loader('habana_frameworks', loader=None)
+    habana_frameworks.__spec__ = spec
+    habana_frameworks.torch = types.ModuleType(  # type: ignore
+        'habana_frameworks.torch')
+    habana_frameworks.torch.core = types.ModuleType(  # type: ignore
+        'habana_frameworks.torch.core')
+
+    sys.modules['habana_frameworks'] = habana_frameworks
+    sys.modules['habana_frameworks.torch'] = habana_frameworks.torch
+    sys.modules['habana_frameworks.torch.core'] = habana_frameworks.torch.core
+
+    habana_frameworks.torch.utils = types.ModuleType(  # type: ignore
+        'habana_frameworks.torch.utils')
+    habana_frameworks.torch.utils.internal = types.ModuleType(  # type: ignore
+        'habana_frameworks.torch.utils.internal')
+    sys.modules[
+        'habana_frameworks.torch.utils'] = habana_frameworks.torch.utils
+    sys.modules['habana_frameworks.torch.utils.internal'] = (
+        habana_frameworks.torch.utils.internal)
+
+    habana_frameworks.torch.internal = types.ModuleType(  # type: ignore
+        'habana_frameworks.torch.internal')
+    habana_frameworks.torch.internal.bridge_config = types.ModuleType(  # type: ignore
+        'habana_frameworks.torch.internal.bridge_config')
+    sys.modules[
+        'habana_frameworks.torch.internal'] = habana_frameworks.torch.internal
+    sys.modules['habana_frameworks.torch.internal.bridge_config'] = (
+        habana_frameworks.torch.internal.bridge_config)
+
+    torch.hpu = types.ModuleType('torch.hpu')  # type: ignore
+    sys.modules['torch.hpu'] = torch.hpu
+
+    habana_frameworks.torch.core.mark_step = lambda: print(  # type: ignore
+        'calling mark_step')
+    habana_frameworks.torch.utils.internal.is_lazy = lambda: print(  # type: ignore
+        'calling is_lazy')
+    torch.hpu.synchronize = lambda: print('calling synchronize'  # type: ignore
```

Comment:
How does this correspond to definitions in _migrate_to_cpu()?

Reply:
Removed.